src/playground/public/assets/js/app.ts

```diff
@@ -32,9 +32,15 @@ function registerAskScriptEditor(
   askScriptServerUrl: string
 ) {
   const editor = ace.edit(editorElementId);
+  let dirty = false;
   editor.setTheme('ace/theme/twilight');
   editor.session.setMode('ace/mode/javascript');
   editor.session.setOption('useWorker', false);
+  editor.session.on('change', () => {
+    if (dirty) return;
+    playgroundIsDirty(!dirty);
+    dirty = true;
+  });
```

Comment:
I think that now when we have `isDirty` check, we can just attach `onbeforeunload` once for all without the need for the extra `playgroundIsDirty` function.

```suggestion
  });
  
  window.onbeforeunload = function (e: any) {
    if (isDirty) {
      e.preventDefault();
      e.returnValue = '';
    }
  };
```